### ISSUE TITLE
[iOS] random crash on annotation selection

### ISSUE THREAD

**Issue by RomainQuidet** [mapbox/mapbox-gl-native#3284]
Use the demo app, long press to add dropped annotations. After 3/4 annotations added, a selection went wrong and crashed:

```
2015-12-14 17:47:40.186 Mapbox GL[54555:1345769] *** Assertion failure in -[MGLMapView handleSingleTapGesture:], /Users/mappy/Documents/mapbox-gl-native/platform/ios/MGLMapView.mm:1142
2015-12-14 17:47:40.203 Mapbox GL[54555:1345769] *** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'Cannot select nonexistent annotation with tag -252933991'
*** First throw call stack:
(
	0   CoreFoundation                      0x000000010d782e65 __exceptionPreprocess + 165
	1   libobjc.A.dylib                     0x000000010d154deb objc_exception_throw + 48
	2   CoreFoundation                      0x000000010d782cca +[NSException raise:format:arguments:] + 106
	3   Foundation                          0x000000010cc784de -[NSAssertionHandler handleFailureInMethod:object:file:lineNumber:description:] + 198
	4   Mapbox GL                           0x000000010ba3a343 -[MGLMapView handleSingleTapGesture:] + 1507
	5   UIKit                               0x000000010e31ae73 _UIGestureRecognizerSendTargetActions + 153
	6   UIKit                               0x000000010e3174e5 _UIGestureRecognizerSendActions + 162
	7   UIKit                               0x000000010e3154e2 -[UIGestureRecognizer _updateGestureWithEvent:buttonEvent:] + 843
	8   UIKit                               0x000000010e31d9a0 ___UIGestureRecognizerUpdate_block_invoke904 + 79
	9   UIKit                               0x000000010e31d83e _UIGestureRecognizerRemoveObjectsFromArrayAndApplyBlocks + 342
	10  UIKit                               0x000000010e30b101 _UIGestureRecognizerUpdate + 2634
	11  CoreFoundation                      0x000000010d6ae367 __CFRUNLOOP_IS_CALLING_OUT_TO_AN_OBSERVER_CALLBACK_FUNCTION__ + 23
	12  CoreFoundation                      0x000000010d6ae2d7 __CFRunLoopDoObservers + 391
	13  CoreFoundation                      0x000000010d6a3f2b __CFRunLoopRun + 1147
	14  CoreFoundation                      0x000000010d6a3828 CFRunLoopRunSpecific + 488
	15  GraphicsServices                    0x00000001100b4ad2 GSEventRunModal + 161
	16  UIKit                               0x000000010de32610 UIApplicationMain + 171
	17  Mapbox GL                           0x000000010b5b3dff main + 111
	18  libdyld.dylib                       0x00000001106c692d start + 1
)
libc++abi.dylib: terminating with uncaught exception of type NSException
```

**Comment by 1ec5** [comment]
I haven’t encountered this crash yet, but it seems to indicate that `AnnotationManager::getPointAnnotationsInBounds()` returned a bogus annotation ID.

**Comment by 1ec5** [comment]
The assertion message was using the wrong format specifier. I corrected that in c3fe62fd3a4a3d54f1661a145f7d1d4e9a86f226; I think that’ll help us understand (for instance) how many annotations mbgl thinks got added.

**Comment by maciekish** [comment]
We have the same issue. We can provide steps to recreate in a production app if that helps at all.

**Comment by maciekish** [comment]
@1ec5 It happens in [hitta.se](https://geo.itunes.apple.com/us/app/hitta.se/id284963764?mt=8) If you zoom in to the hospital in the video, tap the annotation to reveal the callout, long press on the callout to drop a pin and then tap the original annotation it crashes with the same message as the original poster. Video http://s.swic.name/eyb4

**Comment by Zakay** [comment]
Steps to reproduce:

1. Drop pin 1 anywhere
2. Drop pin 2 anywhere
3. Tap on pin 1
4. Drop pin 3 anywhere
5. Tap on pin 1

**Comment by 1ec5** [comment]
Fixed in #3881. The fix will be included in 3.1.1, which will be out shortly.
